client/chain-spec/src/chain_spec.rs

```diff
@@ -378,6 +402,49 @@ where
 	fn set_storage(&mut self, storage: Storage) {
 		self.genesis = GenesisSource::Storage(storage);
 	}
+
+	fn set_hardcoded_sync(&mut self, hardcoded_sync: SerializableHardcodedSync) {
+		ChainSpec::set_hardcoded_sync(self, hardcoded_sync)
+	}
+}
+
+/// Hardcoded infomation that allows light clients to sync quickly.
+pub struct HardcodedSync<Block: BlockT> {
+	/// The header of the best finalized block.
+	pub header: <Block as BlockT>::Header,
+	/// A list of all CHTs in the chain.
+	pub chts: Vec<<Block as BlockT>::Hash>,
+}
+
+impl<Block: BlockT> HardcodedSync<Block> {
+	/// Convert into a `SerializableHardcodecSync`.
+	pub fn to_serializable(&self) -> SerializableHardcodedSync {
+		use codec::Encode;
+
+		SerializableHardcodedSync {
+			header: StorageData(self.header.encode()),
+			chts: self.chts.iter().map(|hash| StorageData(hash.encode())).collect(),
+		}
+	}
+
+	/// Convert from a `SerializableHardcodecSync`.
+	pub fn from_serializable(serialized: &SerializableHardcodedSync) -> Result<Self, codec::Error> {
+		Ok(Self {
+			header: codec::Decode::decode(&mut &serialized.header.0[..])?,
+			chts: serialized.chts.iter()
+				.map(|cht| codec::Decode::decode(&mut &cht.0[..]))
+				.collect::<Result<_, _>>()?,
+		})
+	}
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
+/// The serializable form of `HardcodedSync`. Created using `HardcodedSync::serialize`.
+pub struct SerializableHardcodedSync {
+	header: StorageData,
+	chts: Vec<StorageData>,
```

Comment:
I believe that `StorageData` is the best way to serialize these, but there could be a better alternative.